src/rules/curlyRule.ts

```diff
@@ -129,12 +140,64 @@ class CurlyWalker extends Lint.AbstractWalker<Options> {
         return ts.forEachChild(sourceFile, cb);
     }
 
-    private checkStatement(statement: ts.Statement, node: ts.Node, childIndex: number, end = statement.end) {
+    private checkStatement(statement: ts.Statement, node: ts.IterationStatement | ts.IfStatement, childIndex: number, end = statement.end) {
+        const sameLine = isSameLine(this.sourceFile, statement.pos, statement.end);
         if (statement.kind !== ts.SyntaxKind.Block &&
-            !(this.options.ignoreSameLine && isSameLine(this.sourceFile, statement.pos, statement.end))) {
+            !(this.options.ignoreSameLine && sameLine)) {
             const token = node.getChildAt(childIndex, this.sourceFile);
             const tokenText = ts.tokenToString(token.kind)!;
-            this.addFailure(token.end - tokenText.length, end, Rule.FAILURE_STRING_FACTORY(tokenText));
+            this.addFailure(
+                token.end - tokenText.length, end, Rule.FAILURE_STRING_FACTORY(tokenText),
+                this.addBraceReplacement(statement, node, sameLine));
+        }
+    }
+
+    /** Generate the necessary replacement to add braces to a statement that needs them. */
+    private addBraceReplacement(statement: ts.Statement, node: ts.IterationStatement | ts.IfStatement, sameLine: boolean) {
```

Comment:
better name it `createMissingBraceFix` or similar

Reply:
Done